### ISSUE TITLE
create-new-task doesn't work.

### ISSUE THREAD

**Issue by TigerDevloper** [lehel-twilio/plugin-dialpad#14]
I tried to test it on my flex application. But I don't know how to create a function for endpoint "/create-new-task".
Can you tell me how to write the function so that I can test it?

**Comment by danruss** [comment]
The README file references the function code that is available here: https://github.com/lehel-twilio/plugin-dialpad/tree/master/src/functions

Each function needs to be created and named consistent with the filename (minus the .js extension)

**Comment by TigerDevloper** [comment]
Well, what I meant is  Twilio Function, not javascript function.
To make your plugin work, there should be a Twilio function.
And I want to know how to write in that Twilio function body.
Can you show me some screenshots on the Twilio functions?

**Comment by danruss** [comment]
You should start with the [Twilio Functions docs](https://www.twilio.com/docs/runtime/functions#how-it-works).  Once you get a sense how they work and where to create them, you can create each function from the GitHub linked above.  Be sure you create the functions in the same Twilio project as your Flex instance.

**Comment by TigerDevloper** [comment]
Well, I know how and where to create the task. So frankly speaking, I tried to make an outbound call by myself.
So I created a task and then tried to create a conference but I got an error saying that "The conference instruction can only be issued on a task that was created using the <Enqueue> TwiML verb."
That's where I got stuck.

At that time, I found your plugin by chance and tried to use it on my flex project, but failed because of the twilio functions.

**Comment by vpatidar89** [comment]
Hello all
I am getting an error when I click on the dailer button in function "/create-new-task"
"message": "Cannot find module 'axios'"

How can I import dependency in Twilio Flex?

I import Axios in a plugin and built it and place plugin.js to Twilio flex assets.

**Comment by lehel-twilio** [comment]
In the Twilio Console, go to Runtime -> Functions -> Configure and add it to the dependencies section on the bottom.

**Comment by vpatidar89** [comment]
That's perfect what i was looking for.

But I found a new issue

The requested resource
/Workspaces/WKcb43107f7c0ecdb3bccd75d25ce6a66b/Tasks was not found


Which type of task do i need to create here?


Thanks, n Regards,

Vishal Patidar
Dynamics CRM Developer
MoreYeahs Inc.

**Comment by lehel-twilio** [comment]
Your workspace sid is incorrect. It starts with WSxxxxx

**Comment by vpatidar89** [comment]
U r savior Rex.
Thanks for making my day. I am able to make an outbound call Thanks.


Thanks, n Regards,

Vishal Patidar
Dynamics CRM Developer
MoreYeahs Inc.

**Comment by vpatidar89** [comment]
hi, rex,

Can you help me with this? I was able to make outbound calls but only once.
I am able to make outbound but getting ERROR - 11200 HTTP retrieval failure.

Is this due to the internal issue?

Thanks, n Regards,

Vishal Patidar
Dynamics CRM Developer
MoreYeahs Inc.

**Comment by vpatidar89** [comment]
hi, rex,

Is there an internal issue going on?
i am getting 404 from functions as well as task post url
https://gray-gazelle-2903.twil.io/create-new-task
https://gray-gazelle-2903.twil.io/agent-outbound-join?taskSid=WT0866509034f7868fd694fa737b01fdb4


Thanks, n Regards,

Vishal Patidar
Dynamics CRM Developer
MoreYeahs Inc.

**Comment by lehel-twilio** [comment]
If you navigate to these functions in the Twilio Console, and run a new example, you will get debug logs at the bottom of the page. Can you do that and post the results here?

**Comment by vpatidar89** [comment]
I am gettinig this error:
Error - 11200
Application
Properties
DEBUG EVENT SIDNO6d8986a5ccc66cfce27dd2159e712390SERVICE SID
RESOURCE SIDCA72d816c0f7db031c28221a5674cbb1f1
<https://www.twilio.com/console/voice/logs/calls/CA72d816c0f7db031c28221a5674cbb1f1>
TIMESTAMP2019-07-17 05:14:50 UTC
MESSAGEAn attempt to retrieve content from
https://gray-gazelle-2903.twil.io/agent-outbound-join?taskSid=WTecfc132281d3406dcf173fa09212ce43
returned the HTTP status code 404
HTTP retrieval failure
Possible Causes

   - Web server returned a 4xx or 5xx HTTP response to Twilio
   - Misconfigured Web Server
   - Network disruptions between Twilio and your web server
   - No Content-Type header attached to response
   - Content-Type doesn’t match actual content, e.g. an MP3 file that is
   being served with Content-Type: audio/x-wav, instead of Content-Type:
   audio/mpeg


Possible Solutions

   - Double check that your TwiML URL does not return a 4xx or 5xx error
   - Make certain that the URL does not perform a 302 redirect to an
   invalid URL
   - Confirm the URL requested is not protected by HTTP Auth
   - Make sure your web server allows HTTP POST requests to static
   resources (if the URL refers to .xml or .html files)
   - Verify your web server is up and responsive
   - Check to see that the URL host is not a private or local IP address
   - Verify the ping times and packet loss between your web server and
   www.twilio.com


Error Description

There was a failure attempting to retrieve the contents of this URL.

Request Inspector
- Collapse All
POST
https://gray-gazelle-2903.twil.io/agent-outbound-join
2019-07-17 05:14:50 UTC
404
Request
URL

https://gray-gazelle-2903.twil.io/agent-outbound-join

ParametersShow Raw
ApiVersion "2010-04-01"
Called "client:v_2DvPatidar"
CallStatus "in-progress"
From "+8474257284327"
CallerCountry "VN"
Direction "outbound-api"
AccountSid "ACc56e44dca0b309162b120e668be394a6"
CallerCity ""
CallerState "Tra Vinh"
FromCountry "VN"
Caller "+8474257284327"
FromCity ""
To "client:v_2DvPatidar"
CallSid "CA72d816c0f7db031c28221a5674cbb1f1"
FromZip ""
CallerZip ""
FromState "Tra Vinh"
Message TextShow Raw
Msg "An attempt to retrieve content from
https://gray-gazelle-2903.twil.io/agent-outbound-join?taskSid=WTecfc132281d3406dcf173fa09212ce43
returned the HTTP status code 404"
msg "An attempt to retrieve content from
https://gray-gazelle-2903.twil.io/agent-outbound-join?taskSid=WTecfc132281d3406dcf173fa09212ce43
returned the HTTP status code 404"
sourceComponent "12000"
ErrorCode "11200"
httpResponse "404"
LogLevel "ERROR"
url "
https://gray-gazelle-2903.twil.io/agent-outbound-join?taskSid=WTecfc132281d3406dcf173fa09212ce43
"
Response
Headers
X-Cache MISS from Twilio-Cache
X-Shenanigans none
X-Cache-Lookup MISS from Twilio-Cache:3128
X-Twilio-WebhookRetriable false
Content-Length 9
Date Wed, 17 Jul 2019 05:14:50 GMT
Content-Type text/plain; charset=utf8
Body
*No body.*






Vishal Patidar
Dynamics CRM Developer
MoreYeahs Inc.

**Comment by vpatidar89** [comment]
Please ignore above mail.
I didn't create functions that's why it is happening. now it is resolved.



Thanks, n Regards,

Vishal Patidar
Dynamics CRM Developer
MoreYeahs Inc.

**Comment by vpatidar89** [comment]
HI

Now I am getting two error

[image: image.png]


when I click on 400 it is not opening and loading.

second error detail:
Properties
DEBUG EVENT SIDNO005c6bb5a329f8dc3df1e9161d17635eSERVICE SID
RESOURCE SIDN/ATIMESTAMP2019-07-17 06:57:38 UTC
TaskRouter->Call
Error - 40113Error Making Outgoing Call - Call InstructionPossible Causes

   - Un-routable address provided for "to" field.
   - Invalid caller-id passed to "from" field.
   - Poorly formed URL passed to "url" field.
   - Internal Twilio API error.

Possible Solutions

   - Make certain "to" contains a routable phone number, client URI or SIP
   URI.
   - Make certain "to" is to a country you have permissions set to call.
   - Make certain "from" contains an acceptable caller ID for the chosen
   "to" address.
   - Make certain "url" is a valid HTTP/HTTPS URL.

Example

{
	"instruction": "call",
	"to": "client:alice",
	"from": "bob",
	"url": "https://example.com/twiml"
}

Request Inspector
- Collapse All
2019-07-17 06:57:38 UTC
400
Request
URL

Parameters
*No Parameters*
Message TextShow Raw
msg "The call instruction encountered an error when requesting an outgoing
call be made from the API. Code: [21212] - From is not a valid phone
number: Enter a Twilio Number here - ReservationSid:
WR02ffd1188c647b91cad2524db9bedec2"
httpResponse "400"
url ""
Response
Headers
Content-Length 140
Date Wed, 17 Jul 2019 06:57:38 GMT
Content-Type application/json;charset=utf-8
BodyShow Raw

{
    "message": "From is not a valid phone number: Enter a Twilio Number here",
    "code": 21212,
    "user_error": false,
    "http_status_code": -1,
    "params": {}
}






Vishal Patidar
Dynamics CRM Developer
MoreYeahs Inc.

**Comment by lehel-twilio** [comment]
Do the workers have a "phone" attribute? This will get passed down to the function.

**Comment by vpatidar89** [comment]
Yes it's working now. Just one question I want to make call programmically
from dialpad.js


I want to create a function with phone number parameters. On calling it
will make a same outbound call as it happens with dialpad.

**Comment by vpatidar89** [comment]
Ignore above mail I read the detail in the documentation.
Thanks for your support.

I will get back to you if I have any queries




Thanks, n Regards,

Vishal Patidar
Dynamics CRM Developer
MoreYeahs Inc.
